File: serverHR/src/main/java/hr/server/serverhr/entities/Training.java
package hr.server.serverhr.entities;

import lombok.*;

import javax.persistence.*;
import java.io.Serializable;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Training implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Setter(AccessLevel.NONE)
    private Long id;

    private String name;
    private String description;
    private LocalDate startDate;
    private LocalDate endDate;

    @ManyToMany
    @JoinTable(
            name = "training_employee",
            joinColumns = @JoinColumn(name = "training_id"),
            inverseJoinColumns = @JoinColumn(name = "idEmployee")
    )
    private List<Employee> attendees;

}

File: clientHR/src/app/models/training.ts
import { Employee } from "./employee";

export class Training {
    id!: number; 
    name!: string; 
    description!: string; 
    startDate!: string; // Start date in ISO string format
    endDate!: string; // End date in ISO string format
    attendees!: Set<Employee>; 
}
